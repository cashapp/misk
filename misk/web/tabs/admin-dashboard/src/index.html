<!DOCTYPE html>
<html>

<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
  <meta http-equiv="X-UA-Compatible" content="ie=edge" />
  <link rel="shortcut icon" type="image/x-icon" href="/static/favicon.ico" />
</head>

<body>
  <div id="admin-dashboard"></div>

  <!-- Misk Libraries -->
  <link rel="stylesheet" type="text/css" href="/@misk/common/styles.css" />
  <script type="text/javascript" src="/@misk/common/vendors.js" preload></script>
  <script type="text/javascript" src="/@misk/common/common.js" preload></script>
  <script type="text/javascript" src="/@misk/core/core.js" preload></script>
  <script type="text/javascript" src="/@misk/simpleredux/simpleredux.js" preload></script>

  <!-- Tab Admin Dashboard -->
  <script type="text/javascript" src="/_admin/tab_admin-dashboard.js"></script>
</body>

</html>
