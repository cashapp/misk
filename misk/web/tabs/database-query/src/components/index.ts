export * from "./QuantityButton"
export * from "./CommonComponents"
export * from "./RequestResponseContentTypes"
export * from "./SkeletonWebActionsComponent"
export * from "./StatusTagComponent"
