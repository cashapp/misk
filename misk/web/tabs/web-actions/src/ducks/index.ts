import {
  dispatchSimpleRedux,
  IDispatchSimpleRedux,
  ISimpleReduxImmutableState,
  ISimpleReduxState,
  SimpleReduxReducer,
  SimpleReduxSaga,
  simpleRootSelector,
  watchSimpleReduxSagas
} from "@misk/simpleredux"
import {
  connectRouter,
  LocationChangeAction,
  RouterState
} from "connected-react-router"
import { History } from "history"
import { AnyAction, combineReducers, Reducer } from "redux"
import { all, fork } from "redux-saga/effects"
import {
  dispatchWebActions,
  IDispatchWebActions,
  IWebActionsImmutableState,
  IWebActionsState,
  watchWebActionsSagas,
  WebActionsReducer
} from "./webActions"
export * from "./webActions"

/**
 * Redux Store State
 */
export interface IState {
  router: Reducer<RouterState, LocationChangeAction>
  simpleRedux: ISimpleReduxState
  webActions: IWebActionsState
  webActionsRaw: IWebActionsImmutableState
}

/**
 * Dispatcher
 */
export interface IDispatchProps
  extends IDispatchSimpleRedux,
    IDispatchWebActions {}

export const rootDispatcher: IDispatchProps = {
  ...dispatchSimpleRedux,
  ...dispatchWebActions
}

/**
 * State Selectors
 */
export const rootSelectors = (state: IState) => ({
  router: state.router,
  simpleRedux: simpleRootSelector<IState, ISimpleReduxImmutableState>(
    "simpleRedux",
    state
  ),
  webActions: simpleRootSelector<IState, IWebActionsImmutableState>(
    "webActions",
    state
  ),
  webActionsRaw: state.webActionsRaw
})

/**
 * Reducers
 */
export const rootReducer = (history: History): Reducer<any, AnyAction> =>
  combineReducers({
    router: connectRouter(history),
    simpleRedux: SimpleReduxReducer,
    webActions: WebActionsReducer,
    webActionsRaw: WebActionsReducer
  })

/**
 * Sagas
 */
export function* rootSaga(): SimpleReduxSaga {
  yield all([fork(watchWebActionsSagas), fork(watchSimpleReduxSagas)])
}

/**
 * Map Dispatch/State to Props
 */
export const mapStateToProps = (state: IState) => rootSelectors(state)

export const mapDispatchToProps: IDispatchProps = {
  ...rootDispatcher
}
