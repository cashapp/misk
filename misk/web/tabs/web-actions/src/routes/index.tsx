import * as React from "react"
import { Route, Switch } from "react-router"
import { TabContainer as OldTabContainer } from "../containers"
import TabContainer from "../rewrite/TabContainer"

const routes = (
  <div>
    <Switch>
      <Route path="/_admin/web-actions/" component={OldTabContainer} />
      <Route path="/_admin/web-actions-beta/" component={TabContainer} />
      {/* Do not include a Route without a path or it will display during on all tabs */}
    </Switch>
  </div>
)

export default routes
