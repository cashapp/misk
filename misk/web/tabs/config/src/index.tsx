///<reference types="webpack-env" />
import { IWindow } from "@misk/common"
import { connectRouter, routerMiddleware } from "connected-react-router"
import { createBrowserHistory } from "history"
import * as React from "react"
import * as ReactDOM from "react-dom"
import { AppContainer } from "react-hot-loader"
import { Provider } from "react-redux"
import { applyMiddleware, compose, createStore } from "redux"
import createSagaMiddleware from "redux-saga"
import App from "./App"
import { rootReducer, rootSaga } from "./ducks"

export * from "./components"
export * from "./containers"

const tabSlug = "config"

const Window = window as IWindow

Window.Misk.History = Window.Misk.History || createBrowserHistory()
const history = Window.Misk.History
const sagaMiddleware = createSagaMiddleware()

const composeEnhancer: typeof compose = Window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
const store = createStore(
  connectRouter(history)(rootReducer),
  composeEnhancer(
    applyMiddleware(
      sagaMiddleware,
      routerMiddleware(history),
    ),
  ),
)

/**
 * Starts the rootSaga which forks off instances of all sagas used to receive and process actions as they are dispatched (./sagas/index.ts)
 */
sagaMiddleware.run(rootSaga)

const render = () => {
  ReactDOM.render(
    <AppContainer>
      <Provider store={store}>
        <App history={history} />
      </Provider>
    </AppContainer>,
    document.getElementById(tabSlug)
  )
}

render()

// Hot reloading
if (module.hot) {
  // Reload components
  module.hot.accept("./App", () => {
    render()
  })

  // Reload reducers
  module.hot.accept("./ducks", () => {
    store.replaceReducer(connectRouter(history)(rootReducer))
  })
}