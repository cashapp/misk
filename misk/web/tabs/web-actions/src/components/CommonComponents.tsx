/** @jsx jsx */
import {
  Collapse,
  Colors,
  Icon,
  MenuItem,
  Tag,
  Tooltip
} from "@blueprintjs/core"
import { IconNames } from "@blueprintjs/icons"
import { css, jsx } from "@emotion/core"
import { HTTPMethodIntent, WrapTextContainer } from "@misk/core"
import {
  ISimpleReduxState,
  onChangeToggleFnCall,
  onClickFnCall,
  simpleSelectorGet
} from "@misk/simpleredux"
import copy from "copy-to-clipboard"
import HTTPMethod from "http-method-enum"
import { connect } from "react-redux"
import { IDispatchProps, mapDispatchToProps, mapStateToProps } from "../ducks"

export const cssFloatLeft = css`
  float: left;
  margin: 5px 10px 0 0;
`

export const cssFloatRight = css`
  float: right;
  margin: 5px 0 0 10px;
`

export const cssCodeTag = css`
  font-family: monospace;
`

export const cssHeader = css`
  display: inline-block;
`

export const cssColumn = css`
  flex-grow: 1;
  flex-basis: 0;
  min-width: 320px;
`

export const cssButton = css`
  line-height: normal;
  text-transform: inherit;
`

export const cssTooltip = css`
  span.bp3-popover-target {
    display: inherit !important;
  }
`

/**
 * Renders HTTP Method tags for each Web Action card
 */
export const MethodTag = (props: { method: HTTPMethod }) => (
  <Tag css={cssFloatRight} large={true} intent={HTTPMethodIntent[props.method]}>
    {props.method}
  </Tag>
)

export const cssMetadataMenu = css`
  li {
    margin-bottom: 0;
  }
`

export const cssWrapTextArea = css`
  white-space: pre-line !important;
`

const definedOrDefault = (item: any, booleanIfUndefined: boolean) =>
  item !== undefined ? item : booleanIfUndefined

/**
 * Single cell component for a piece of metadata
 * @param props
 *  * content: primary content that is displayed (ie. eng, service-owners)
 *  * label: right aligned faded text that is a label for the data (ie. Roles)
 *  * tooltip: text displayed in tooltip when content is hovered over.
 *      Separate definition from content in the case that a truncated length
 *      is displayed but the full length of content is displayed in tooltip
 */
export const Metadata = (props: {
  content: string | JSX.Element
  label?: string
  labelElement?: JSX.Element
  onClick?: (event: any) => void
  tooltip?: string | JSX.Element
}) => {
  if (props.tooltip) {
    return (
      <MenuItem
        {...props}
        text={
          <Tooltip css={cssTooltip} content={props.tooltip} lazy={true}>
            {props.content}
          </Tooltip>
        }
      />
    )
  } else {
    return <MenuItem {...props} text={props.content} />
  }
}

/**
 * Metadata pre-filled with content to support a click-to-copy-to-clipboard flow
 */
export const MetadataCopyToClipboard = (props: {
  clipboardLabelElement?: boolean
  content?: string | JSX.Element
  data: any
  description?: string
  label?: string
  labelElement?: JSX.Element
  onClick?: (event: any) => void
  tooltip?: string | JSX.Element
}) => {
  const description = props.description && `${props.description} `
  const content: string | JSX.Element =
    props.content || `Click to Copy ${description}to Clipboard`
  const labelElement: JSX.Element =
    props.labelElement ||
    (definedOrDefault(props.clipboardLabelElement, true) && (
      <Tooltip css={cssTooltip} content={"Click to Copy to Clipboard"}>
        <Icon icon={IconNames.CLIPBOARD} />
      </Tooltip>
    ))
  const safeData =
    typeof props.data === "string" ? props.data : JSON.stringify(props.data)
  return (
    <Metadata
      content={content}
      label={props.label}
      labelElement={labelElement}
      onClick={onClickFnCall(copy, safeData)}
      tooltip={props.tooltip}
    />
  )
}

/**
 * MetadataCollapse is a generic container for all Metadata
 * It handles
 *  * Sliding out to reveal child content when clicked
 *  * Automatically showing overflow content below when clicked
 *  * Displaying a list of given content below when clicked
 *  * In most cases all content is "click-to-copy" enabled
 *
 * @param props : includes same props as Metadata with a few additional
 *  * children: any components to display when the Metadata is clicked
 *  * tag: string to use in @misk/SimpleRedux/SimpleRedux to register Metadata clicks
 *  * IDispatchProps: include connected dispatch object from parent container
 *      Provides access to @misk/SimpleRedux/SimpleRedux input handlers
 */

const UnconnectedMetadataCollapse = (
  props: {
    clipboardLabelElement?: boolean
    children?: any
    content?: string | string[] | JSX.Element | JSX.Element[]
    countLabel?: boolean
    data?: string
    isOpen?: boolean
    label?: string
    labelElement?: JSX.Element
    simpleRedux: ISimpleReduxState
    tag: string
    text?: string | JSX.Element
    tooltip?: string | JSX.Element
  } & IDispatchProps
) => {
  const content = Array.isArray(props.content) ? props.content : [props.content]
  const collapseIcon = simpleSelectorGet(props.simpleRedux, [
    props.tag,
    "data"
  ]) ? (
    <Icon icon={IconNames.CARET_DOWN} />
  ) : (
    <Icon icon={IconNames.CARET_RIGHT} />
  )
  return (
    <div>
      <Metadata
        content={
          <span>
            {props.children || content.length > 0 ? (
              collapseIcon
            ) : (
              <Icon color={Colors.LIGHT_GRAY1} icon={IconNames.CARET_RIGHT} />
            )}
            {props.text || props.children || content.join(", ")}
          </span>
        }
        data-testid={"metadata-collapse"}
        label={
          props.children || !props.countLabel
            ? props.label
            : `${props.label} (${content.length})`
        }
        labelElement={props.children ? props.labelElement : null}
        onClick={onChangeToggleFnCall(
          props.simpleMergeToggle,
          props.tag,
          props.simpleRedux
        )}
      />
      <Collapse
        isOpen={simpleSelectorGet(props.simpleRedux, [props.tag, "data"])}
      >
        {props.children
          ? props.children
          : content.map(c => (
              <MetadataCopyToClipboard
                clipboardLabelElement={definedOrDefault(
                  props.clipboardLabelElement,
                  true
                )}
                content={<WrapTextContainer>{c}</WrapTextContainer>}
                data={props.data || c}
                key={c.toString()}
                labelElement={props.labelElement}
              />
            ))}
      </Collapse>
    </div>
  )
}

export const MetadataCollapse = connect(
  mapStateToProps,
  mapDispatchToProps
)(UnconnectedMetadataCollapse)
