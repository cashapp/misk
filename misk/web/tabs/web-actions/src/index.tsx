import { createApp, createIndex } from "@misk/core"
import * as Ducks from "./ducks"
import routes from "./routes"
export * from "./containers"

createIndex("web-actions", createApp(routes), Ducks)
