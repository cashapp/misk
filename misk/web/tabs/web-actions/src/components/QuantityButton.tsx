/** @jsx jsx */
import { Button, Tooltip } from "@blueprintjs/core"
import { IconName } from "@blueprintjs/icons"
import { jsx } from "@emotion/core"
import { onChangeFnCall } from "@misk/simpleredux"
import { IWebActionInternal } from "../ducks"
import { cssButton, cssTooltip } from "./CommonComponents"

export const QuantityButton = (props: {
  id: string
  action: IWebActionInternal
  changeFn: any
  content: string
  icon: IconName
  oldState: any
}) => (
  <Tooltip css={cssTooltip} content={props.content} lazy={true}>
    <Button
      css={cssButton}
      icon={props.icon}
      onClick={onChangeFnCall(
        props.changeFn,
        props.id,
        props.action,
        props.oldState
      )}
    />
  </Tooltip>
)
