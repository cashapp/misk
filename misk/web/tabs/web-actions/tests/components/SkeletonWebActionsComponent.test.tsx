import React from "react"
import { cleanup, render } from "@testing-library/react"
import { SkeletonText, SkeletonWebActionsComponent } from "../../src/components"

describe("SkeletonWebActionsComponent", () => {
  afterEach(cleanup)
  it("SkeletonText glowing text block used as loading screen", () => {
    const { asFragment } = render(<SkeletonText />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("SkeletonWebActionsComponent single Web Actions Card used as loading screen", () => {
    const { asFragment } = render(<SkeletonWebActionsComponent />)
    expect(asFragment()).toMatchSnapshot()
  })
})
