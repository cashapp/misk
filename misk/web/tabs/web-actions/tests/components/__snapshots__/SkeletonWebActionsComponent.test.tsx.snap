// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonWebActionsComponent SkeletonText glowing text block used as loading screen 1`] = `
<DocumentFragment>
  <span
    class="bp3-skeleton"
  >
    Lorem ipsum
  </span>
</DocumentFragment>
`;

exports[`SkeletonWebActionsComponent SkeletonWebActionsComponent single Web Actions Card used as loading screen 1`] = `
<DocumentFragment>
  .emotion-3 {
  display: inline-block;
}

.emotion-0 {
  float: right;
  margin: 5px 0 0 10px;
}

.emotion-1 {
  float: left;
  margin: 5px 10px 0 0;
}

.emotion-2 {
  float: left;
  margin: 5px 10px 0 0;
  font-family: monospace;
}

.emotion-5 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  min-width: 320px;
}

.emotion-4 li {
  margin-bottom: 0;
}

<div
    class="bp3-card bp3-elevation-0"
  >
    <div
      class="emotion-3"
    >
      <span
        class="bp3-tag bp3-intent-primary bp3-large emotion-0"
      >
        <span
          class="bp3-text-overflow-ellipsis bp3-fill"
        >
          GET
        </span>
      </span>
      <h3
        class="bp3-heading bp3-skeleton emotion-1"
      >
        AnotherSimpleWebAction
      </h3>
      <span
        class="bp3-tag bp3-large emotion-2"
      >
        <span
          class="bp3-text-overflow-ellipsis bp3-fill"
        >
          <span
            class="bp3-skeleton"
          >
            Lorem ipsum
          </span>
        </span>
      </span>
    </div>
    <div
      class="emotion-8"
    >
      <div
        class="emotion-5"
      >
        <ul
          class="bp3-menu emotion-4"
        >
          <li
            class=""
          >
            <a
              class="bp3-menu-item bp3-popover-dismiss"
              content="[object Object]"
              label="Function"
            >
              <div
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                <span
                  class="bp3-skeleton"
                >
                  Lorem ipsum
                </span>
              </div>
              <span
                class="bp3-menu-item-label"
              >
                Function
              </span>
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="bp3-menu-item bp3-popover-dismiss"
              content="[object Object]"
              label="Services"
            >
              <div
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                <span
                  class="bp3-skeleton"
                >
                  Lorem ipsum
                </span>
              </div>
              <span
                class="bp3-menu-item-label"
              >
                Services
              </span>
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="bp3-menu-item bp3-popover-dismiss"
              content="[object Object]"
              label="Roles"
            >
              <div
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                <span
                  class="bp3-skeleton"
                >
                  Lorem ipsum
                </span>
              </div>
              <span
                class="bp3-menu-item-label"
              >
                Roles
              </span>
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="bp3-menu-item bp3-popover-dismiss"
              content="[object Object]"
              label="Access"
            >
              <div
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                <span
                  class="bp3-skeleton"
                >
                  Lorem ipsum
                </span>
              </div>
              <span
                class="bp3-menu-item-label"
              >
                Access
              </span>
            </a>
          </li>
        </ul>
      </div>
      <div
        class="emotion-5"
      >
        <ul
          class="bp3-menu emotion-4"
        >
          <li
            class=""
          >
            <a
              class="bp3-menu-item bp3-popover-dismiss"
              content="[object Object]"
              label="Content Types"
            >
              <div
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                <span
                  class="bp3-skeleton"
                >
                  Lorem ipsum
                </span>
              </div>
              <span
                class="bp3-menu-item-label"
              >
                Content Types
              </span>
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="bp3-menu-item bp3-popover-dismiss"
              content="[object Object]"
              label="Application Interceptors"
            >
              <div
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                <span
                  class="bp3-skeleton"
                >
                  Lorem ipsum
                </span>
              </div>
              <span
                class="bp3-menu-item-label"
              >
                Application Interceptors
              </span>
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="bp3-menu-item bp3-popover-dismiss"
              content="[object Object]"
              label="Network Interceptors"
            >
              <div
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                <span
                  class="bp3-skeleton"
                >
                  Lorem ipsum
                </span>
              </div>
              <span
                class="bp3-menu-item-label"
              >
                Network Interceptors
              </span>
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="bp3-menu-item bp3-popover-dismiss"
              content="[object Object]"
              label="Send a Request"
            >
              <div
                class="bp3-text-overflow-ellipsis bp3-fill"
              >
                <span
                  class="bp3-skeleton"
                >
                  Lorem ipsum
                </span>
              </div>
              <span
                class="bp3-menu-item-label"
              >
                Send a Request
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
